cli/command/stack/deploy_composefile.go

```diff
@@ -321,7 +321,16 @@ func deployServices(
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
 			}
-
+			if !updateOpts.QueryRegistry {
+				serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
+
+				if service.Spec.TaskTemplate.Placement != nil {
+					if serviceSpec.TaskTemplate.Placement == nil {
+						serviceSpec.TaskTemplate.Placement = &swarm.Placement{}
+					}
+					serviceSpec.TaskTemplate.Placement.Platforms = service.Spec.TaskTemplate.Placement.Platforms
```

Comment:
I kept the commit separate but if you check the ServiceUpdate function (sorry, typing on phone so can't link easily); you'll see that it's closely related to the issue;